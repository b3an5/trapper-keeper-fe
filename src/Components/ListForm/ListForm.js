import React, { Component } from 'react'
import PropTypes from 'prop-types'
import checkboxHover from '../../images/completed-hover-icon.svg'

class ListForm extends Component {
  constructor() {
    super()

    this.state = {
      text: '',
    }
  }

  componentDidMount() {
    if (this.props.textValue !== '') {
      this.setState({ text: this.props.textValue })
    }
  }

  handleChange = event => {
    this.setState({ text: event.target.value })
  }

  handleSubmit = async event => {
    event.preventDefault()
    await this.props.setList(this.state.text, this.props.index, this.props.id)
  }

  handleDelete = () => {
    this.props.deleteListItem(this.props.id)
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <button className="form-complete-btn complete-btn">
          {/* // onClick={() => this.props.toggleCompletedLi(li)}> */}
          <img src={checkboxHover} alt="" className="checkbox-hover" />
        </button>
        <input
          className="list-item-input"
          type="text"
          placeholder="add new"
          name="listItem"
          value={this.state.text}
          onChange={this.handleChange}
        />
        <button
          onClick={this.handleDelete}
          className="lf-delete-btn round-btn btn"
        >
          x
        </button>
      </form>
    )
  }
}
ListForm.propTypes = {
  setList: PropTypes.func.isRequired,
}

export default ListForm
