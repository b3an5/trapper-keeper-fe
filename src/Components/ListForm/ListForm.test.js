import React from 'react';
import { shallow } from 'enzyme';
import ListForm from './ListForm'

describe('ListForm', () => {
  let wrapper
  let defaultState = {
    text: ''
  }
  const mockSetList = jest.fn();
 
  const mockEvent = {
    target: { value: 'Test Item'},
    preventDefault: () => { }
  }

  const mockState = { text: 'Test Item'};
  const mockIndex = 0

  beforeEach(() => {
    wrapper = shallow(<ListForm 
                        setList={ mockSetList }
                        index={ 0 }
                      />)
    });

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual(defaultState);
  });

  it('should set state when handleChange is called', () => {
    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state().text).toEqual(mockState.text);
  });

  it('should call setList when handleSubmit is called', () => {    
    wrapper.instance().handleChange(mockEvent);
    wrapper.instance().handleSubmit(mockEvent);

    expect(mockSetList).toHaveBeenCalled();
    expect(mockSetList).toHaveBeenCalledWith(mockState.text, mockIndex )
  });

  it('should match component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
});