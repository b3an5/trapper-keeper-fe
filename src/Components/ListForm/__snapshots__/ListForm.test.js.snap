// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListForm should match component snapshot 1`] = `ShallowWrapper {}`;
